So this test is failed on Travis twice a row, so it could be legit?

```
[error] DEBUGME: Clearing errors: ArrayBuffer()
[error] DEBUGME: Logging: /tmp/sbt_a3ae04a6/semantic-errors/src/main/scala/bad.scala:2:Error: type mismatch;
[error]  found   : Boolean(false)
[error]  required: Int
[info] [success] Total time: 1 s, completed Jul 16, 2015 12:04:50 PM
[info] [info] Defining *:compileOrder
[info] [info] The new value will be used by compile:compile::compileInputs, test:compile::compileInputs
[info] [info] Reapplying settings...
[info] [info] Set current project to semantic-errors (in build file:/tmp/sbt_a3ae04a6/semantic-errors/)
[info] [info] Compiling 1 Scala source and 1 Java source to /tmp/sbt_a3ae04a6/semantic-errors/target/scala-2.10/classes...
[info] [error] /tmp/sbt_a3ae04a6/semantic-errors/src/main/java/bad.java:3: incompatible types
[info] [error] found   : int
[info] [error] required: bad
[info] [error]     public bad foo() { return 1; }
[info] [error]                               ^
[info] [error] 1 error
[info] java.lang.AssertionError: assertion failed: Failed to report any problems!
```

--- a/main/actions/src/main/scala/sbt/Compiler.scala
+++ b/main/actions/src/main/scala/sbt/Compiler.scala
@@ -54,7 +54,7 @@ object Compiler {
   def compilers(instance: ScalaInstance, cpOptions: ClasspathOptions)(implicit app: AppConfiguration, log: Logger): Compilers =
     compilers(instance, cpOptions, None)
 
-  @deprecated("Use `compilers(ScalaInstance, ClasspathOptions, Option[File], IvyConfiguration)`.")
+  @deprecated("Use `compilers(ScalaInstance, ClasspathOptions, Option[File], IvyConfiguration)`.", "0.13.10")
   def compilers(instance: ScalaInstance, cpOptions: ClasspathOptions, javaHome: Option[File])(implicit app: AppConfiguration, log: Logger): Compilers =
     {
       val javac =
@@ -82,7 +82,7 @@ object Compiler {
         def onArgs(f: Seq[String] => Unit): JavaTool = CheaterJavaTool(newJavac, delegate.onArgs(f))
       }
       val scalac = scalaCompiler(instance, cpOptions, ivyConfiguration)
-      new Compilers(scalac, javac)
+      new Compilers(scalac, CheaterJavaTool(javac2, javac))
     }
   @deprecated("Deprecated in favor of new sbt.compiler.javac package.", "0.13.8")
   def compilers(instance: ScalaInstance, cpOptions: ClasspathOptions, javac: sbt.compiler.JavaCompiler.Fork)(implicit app: AppConfiguration, log: Logger): Compilers =

--- a/notes/0.13.10/compiler-interface-specific-sources.md
+++ b/notes/0.13.10/compiler-interface-specific-sources.md
@@ -1,0 +1,11 @@
+
+  [@Duhemm]: http://github.com/Duhemm
+  [2106]: https://github.com/sbt/sbt/pull/2106
+
+### Fixes with compatibility implications
+
+### Improvements
+
+- Fetch the most specific version of the compiler-interface sources available using Ivy [#2106][2106] by [@Duhemm][@Duhemm]
+
+### Bug fixes

